docs/user_guide.md

```diff
@@ -0,0 +1,143 @@
+* `<kbase_developer_username>`: Your username for SSH access. Contact the KBase System Admin team if you do not have access.
+* `@login1.berkeley.kbase.us`: The remote SSH server.   
+   
+### 2. Access Jupyter Notebook:
+   
+Open a web browser and navigate to the following URL:
+
+```
+http://localhost:44041
+```
+This will open the Jupyter Notebook interface running on the remote server.
+
+### 3. Working in the Shared Jupiter Notebook Environment:
+
+Since this is a shared environment, it’s important to create your own directory and keep your work organized.
+
+![user_folders](screen_shots/user_folders.png)
+
+
+We have provided a collection of example notebooks to help you get started. You can find them in the `examples` directory.
+
+### 4. Using Pre-loaded Functions:
+
+To make your development easier, several helper functions and extensions are preloaded in the Jupyter environment.
+
+#### 4.1. Creating a Spark Session:
+
+Use the `get_spark_session` function to create or get a Spark session. 
+
+This function will automatically set up the Spark session with the correct configurations, including setting the 
+master URL and other necessary Spark configurations. By using this function, you ensure that your Spark session is 
+properly configured to interact with the cluster.
+
+##### Example Usage:
+```python
+spark = get_spark_session()
+```
+
+#### 4.2. Displaying DataFrames:
+Use the `display_df` function to display pandas or Spark DataFrames interactively.
+
+The `display_df` function is designed to provide an interactive tabular display of DataFrames within Jupyter Notebooks. 
+It leverages the [itables](https://github.com/mwouts/itables) extension, which enhances the display of data tables by providing features like sorting, 
+filtering, and pagination directly in the notebook interface.
+
+##### Example Usage:
+```python
+display_df(spark.sql(f"SELECT * FROM {namespace}.annotation"))
+```
+![display_df](screen_shots/display_func.png)
+
+### 5. Accessing Data:
+
+#### 5.1. Showing Available Namespaces and Listing Tables:
+To list all namespaces (databases) and display the tables within each namespace, you can use the following code snippet:
+
+```python
+namespaces = spark.sql("SHOW DATABASES").collect()
+
+# List tables within each namespace
+for namespace in namespaces:
+    namespace_name = namespace.namespace
+    print(f"Tables in namespace: {namespace_name}")
+    spark.sql(f"SHOW TABLES IN {namespace_name}").show(50, truncate=False)
+```
+
+### 6. Closing the Spark Session:
+Please remember to close the Spark session when you are done with your work. This will release the resources and 
+prevent any memory leaks.
+
+```python
+spark.stop()
+```
+
+Please be aware that your session will automatically close after `1 hour`. Should you require an extension, simply invoke 
+`get_spark_session()` to initiate a new session.
```

Comment:
this means the prior session is still sitting around wasting resources, right?

Reply:
no. after 1 hr, the prior session will be closed by the timer.

Reply:
But for that hour it is, that's what I'm talking about

Reply:
yea. for the hour the session is open, it's taking resources.

Reply:
I suppose we could keep track of open sessions and close them when a new session is requested but that's probably overkill for now